app/reports/subscription_collection_report.rb

```diff
@@ -0,0 +1,45 @@
+class SubscriptionCollectionReport
+  def initialize(subscriptions)
+    @subscriptions = subscriptions.map do |subscription|
+      SubscriptionReport.new(subscription, columns)
+    end
+  end
+
+  def to_csv
```

Comment:
Yes.

Use `CSV.open` method or other technique that fullfill a file instead of the memory with the report content, doesn't block the request while the report is being generated (use a background job) and send to the frontend a link to the report when it's ready to download